[MM-15296] Migrate "Preference.Save" to Sync by default

IshankGulati opened the pull request:
#### Summary

This PR migrates "Preferences.Save" to sync

#### Ticket Link
Fixes #10720

jespino commented on store/storetest/post_store.go:
```diff
@@ -1276,7 +1276,7 @@ func testPostStoreGetFlaggedPostsForTeam(t *testing.T, ss store.Store, s SqlSupp
 	}
 
-	store.Must(ss.Preference().Save(&preferences))
+	ss.Preference().Save(&preferences)
```
```go
err := ...
require.Nil(t, err)
```

And all the other same cases on this file.

jespino commented on store/storetest/preference_store.go:
```diff
@@ -42,7 +42,7 @@ func testPreferenceSave(t *testing.T, ss store.Store) {
 		},
 	}
-	if count := store.Must(ss.Preference().Save(&preferences)); count != 2 {
+	if count, _ := ss.Preference().Save(&preferences); count != 2 {
```
check the error too

jespino commented on store/storetest/preference_store.go:
```diff
@@ -54,7 +54,7 @@ func testPreferenceSave(t *testing.T, ss store.Store) {
 	preferences[0].Value = "value2a"
 	preferences[1].Value = "value2b"
-	if count := store.Must(ss.Preference().Save(&preferences)); count != 2 {
+	if count, _ := ss.Preference().Save(&preferences); count != 2 {
```
check the error too

jespino commented on store/storetest/preference_store.go:
```diff
@@ -93,7 +93,7 @@ func testPreferenceGet(t *testing.T, ss store.Store) {
 	}
 
-	store.Must(ss.Preference().Save(&preferences))
+	ss.Preference().Save(&preferences)
```
```go
err := ...
require.Nil(t, err)
```

And all the same cases in this file

IshankGulati commented:
@jespino I've addressed all the comments and have tried to keep error handling in the same format as before. Do let me know if any other refactoring would help.

hanzei commented on store/sqlstore/preference_store.go:
```diff
@@ -54,31 +54,25 @@ func (s SqlPreferenceStore) DeleteUnusedFeatures() {
-					break
-				}
-			}
+func (s SqlPreferenceStore) Save(preferences *model.Preferences) (int, *model.AppError) {
```
@jespino Not blocking for this PR, but I'm wondering why this returns an `int`. This is used in the EE code? At least in TE it's not.

jespino replied:
Nope, that code is used only in TE, probably we can remove that int return. But I would prefer to do it in another commit because this one is fully reviewed. @IshankGulati do you want to do it in another PR?

IshankGulati replied:
Yeah sure. I'd raise another PR for that.

jespino commented:
@IshankGulati Do you mind to merge master in this branch?

IshankGulati commented:
@jespino I've updated the PR

hanzei commented on store/storetest/preference_store.go:
```diff
@@ -434,7 +455,9 @@ func testPreferenceCleanupFlagsBatch(t *testing.T, ss store.Store) {
+	require.Nil(t, err)
+	require.Equal(t, 2, count)
 
 	_, err := ss.Preference().CleanupFlagsBatch(10000)
```
This will fix the build error.

```suggestion
	_, err = ss.Preference().CleanupFlagsBatch(10000)
```

IshankGulati replied:
Updated

IshankGulati replied:
I'm not able to debug which tests are breaking from logs. Can you tell me how can I check that?

jespino replied:
To know exactly which error is really causing the test to fail, I normally search for "FAIL" with case sensitivity enabled.

But In this case, I think is a problem with a flaky test. I'm running it again.

jespino commented:
Merged! :tada:
